Makefile

```diff
@@ -136,13 +137,14 @@ $(PROMU):
 	GOOS= GOARCH= go get -u github.com/prometheus/promu
 
 # Always pin dep to the correct version. It changes too often.
-# TODO(bplotka): Install dep without using user's `github.com/golang/dep` package. (Use some tmp place).
 $(DEP):
 	@echo ">> fetching dep from 45be32ba4708aad5e2aa8c86f9432c4c4c1f8da2 revision"
-	@go get -d -u github.com/golang/dep/cmd/dep
-	@cd $(FIRST_GOPATH)/src/github.com/golang/dep && git checkout -f -q 45be32ba4708aad5e2aa8c86f9432c4c4c1f8da2
-	@go install github.com/golang/dep/cmd/dep
-	@mv $(FIRST_GOPATH)/bin/dep $(DEP)
+	@if [ ! -d "$(TMP_GOPATH)/src/github.com/golang/dep" ]; then \
+		GOPATH=$(TMP_GOPATH) go get -d -u github.com/golang/dep/cmd/dep; \
+	fi
```

Comment:
I think that we can fetch the dep when it does not exist. If it has been existed we should just checkout the specify commits. It will save some time.

Reply:
I don't think that will work. I think I am fine with not removing the dir in the beginning, but with this flow this will not work:
1. At some point user do `make deps` and installs deps using on our tmp dir.
2. In future we will change checkout version to some new dep version because of some reason
3. user will fetch new thanos code and do `make deps`. This will try to install versioned dep using same `src/github.com/golang/dep` repo. We will do checkout but since we did not to `go get` (so basically `git fetch`) there won't be such commit in dep git.

Reply:
So can we do to `git fetch` when the `dep` repo exists before checkout the specify commit?

Reply:
yea that would do as well